skbeam/core/cdi.py

```diff
@@ -307,21 +322,24 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
         J. Opt. Soc. Am. A, vol. 20, No. 1, 2003
     """
 
-    diffracted_pattern = np.array(diffracted_pattern)     # diffraction data
+    diffracted_pattern = np.array(diffracted_pattern)  # diffraction data
     diffracted_pattern = np.fft.fftshift(diffracted_pattern)
 
     pi_modulus_flag = pi_modulus_flag.lower()
     real_operation = False
-    if pi_modulus_flag == 'real':
+    if pi_modulus_flag == "real":
         real_operation = True
-    elif pi_modulus_flag == 'complex':
+    elif pi_modulus_flag == "complex":
         real_operation = False
     else:
-        raise ValueError('py_modulus_flag must be one of {"complex",'
-                         'real"} not' '{!r}'.format(pi_modulus_flag))
+        raise ValueError(
```

Comment:
`        raise ValueError('py_modulus_flag must be one of ("complex",'
                         '"real") not' '{!r}'.format(pi_modulus_flag))
`
I think that this is the issue with flake8. The latest version of flake8 seems to make more comprehensive code analysis, so it finds issues in all projects. This is an example when flake8 found a real bug: the code wouldn't work as expected if this branch was hit.

Reply:
(The difference is that {} is replaced by ())

Reply:
```        
            raise ValueError('py_modulus_flag must be one of ("complex", "real")' 
                 'not {!r}'.format(pi_modulus_flag))
```
Better formatted version

Reply:
Ahh.  That makes sense, I was just focused on why it thought things were not in the required format.  I was going to go back in time and start fresh so your message is timely.  

I don't understand why the files I haven't touched were getting flagged.  Do the travis checks check everything?

Reply:
flake8 check the whole project every time you run it, unless you specify the files you want to check. You may have older version of flake8 installed locally, so you may need to update it to see the same result as on Travis, which is always using the latest version.